### ISSUE TITLE
[TW-437] Summary indents incorrectly when using subprojects without a parent project

### ISSUE THREAD

**Issue by taskwarrior** [GothenburgBitFactory/taskwarrior#472]
_Bryan Kam on 2012-08-08T17:03:22Z says:_ 

 As an example:
<pre>
$ task add pro:project1 task1
$ task add pro:project2.subproject2 task2
$ task summary
project1                       1   1 sec       0%
  project2.subproject2         1       -       0%
</pre>

You might say that you should always have a parent task? But is this necessarily the case? I can see cases where I would want all tasks relating to a project divided into subprojects. These would then be indented under the previous (unrelated) project.

**Comment by taskwarrior** [comment]
Migrated metadata:
```
Created: 2012-08-08T17:03:22Z
Modified: 2014-02-09T01:54:30Z
```

**Comment by taskwarrior** [comment]
_Aikido Guy on 2012-08-08T23:38:49Z says:_ 

 I would agree with you... see my comments here http://taskwarrior.org/boards/1/topics/2439

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2012-09-14T23:06:03Z says:_ 

 I have a few thoughts on this:
* -I wonder if "abstract project" would be a good term for this type of project-structure. Just as in C++ one cannot directly create an object of an abstract class and must create an object of a derived class, perhaps for whatever reason it does not make sense to create a task with 'project2'.
* -Does this need to be fixed? If the user doesn't like that indentation, then there are many other delimiters that could be used instead of ".".
* -If this should be fixed, which options should be supported? Are there users who do like the current behavior or would everyone prefer to have no indentation if there does not exist a task with a project "project2" ? Should there be an option to print "project2" as suggested here: http://taskwarrior.org/issues/1106 ? After looking at the code, I think that if there is to be support for printing "project2" then work will have to be done in CmdProjects::execute but if that option does not need to be supported, a cleaner solution is to pass (at)unique@ by reference to the helper function (at)indentProject@ and have the work be done there.

Any thoughts?

**Comment by taskwarrior** [comment]
_David Patrick on 2012-09-15T17:44:32Z says:_ 

 I'll chime in! Let me start by asking two questions;

1) Do you feel that after creating tasks for proj:home.garden and proj:home.admin, 
that the project "home" is implied to exist, even if there are no proj:home tasks right now? 

2) If you said "yes" to the above, would you prefer to have these "implied projects" included in project lists for clarity, even with no tasks? 

if the answer is "yes" to both, how about an .rc setting

 show-empty-parent-projects=yes

or some such thing?
(I think we all agree, but are explaining it differently)
The listing indentation, for real or implied projects, is a separate issue/ setting.

**Comment by taskwarrior** [comment]
_Bryan Kam on 2012-09-19T13:57:14Z says:_ 

 I agree with both of David's points. An option to show "implied projects" would solve the problem.

I think this should be fixed. I see Scott's point that a different delimiter would prevent incorrect indentation, but I'd actually like the subprojects to be indented under the "abstract project" as he calls it. It seems more consistent to support "." for dividing up a project whether or not there is an explicit task for the parent project.

http://taskwarrior.org/issues/1106 seems to be about this same issue. In his example, this would be my ideal output of his data:
<pre>
home
    shopping
    car
work
    project1
    project2
tax
    2011
    2012
</pre>

I think ideally that indentation would work correctly and obviate the repetition of the project in the subproject lines. But perhaps this is a separate request. I'd also be happy with this:
<pre>
home
    home.shopping
    home.car
work
    work.project1
    work.project2
tax
    tax.2011
    tax.2012
</pre>

**Comment by taskwarrior** [comment]
_Dirk Deimeke on 2012-09-19T14:32:24Z says:_ 

 I like the first idea as well.

Spliting it like:
<pre>

home
    shopping
    car
work
    project1
    project2
tax
    2011
    2012
</pre>

**Comment by taskwarrior** [comment]
_David Patrick on 2012-09-19T17:55:51Z says:_ 

 Bryan Kam wrote:
> I'd also be happy with this:

<pre>
home
   home.shopping
   home.car
</pre> 

yuk! REdundant!
:)

**Comment by taskwarrior** [comment]
_Scott Kostyshak on 2012-09-24T21:57:20Z says:_ 

 This should be fixed now in the newest 2.2.0 branch.

Testing is welcome! Please report any problems.
